src/main/java/com/epam/internship/carrental/car/CarController.java

```diff
@@ -0,0 +1,294 @@
+     * <pre>
+     *     Method POST
+     *     URL /api/v1/car/{carId}
+     * </pre>
+     * Sample call /api/v1/car/1
+     *
+     * @param carId         updateable car's ID
+     * @param newCarParams  updateable parameters
+     * @param authorization authorization token from the header of the request
+     * @return ResponseEntity which contains the updated Car
+     * and Response Code 200 on success, or Response Code 403 if the token doesn't match,
+     * or if there is no car with given ID.
+     */
+    @PostMapping(path = "/updatecar/{carId}", consumes = "application/json")
+    public @ResponseBody
+    ResponseEntity<Car> updateCarByGivenParametersWithAuthorization(@PathVariable final Long carId,
+                                                                    @RequestBody final Car newCarParams,
+                                                                    @RequestHeader("Authorization") final String authorization) {
+        return carService.updateCarByGivenParametersWithAuthorization(carId, newCarParams, authorization);
+    }
+
+    /**
+     * Returns all available cars, in a pageable format.
+     * <pre>
+     *     Method GET
+     *     URL /api/v1/free
+     * </pre>
+     * Sample call /api/v1/free?page=0
+     *
+     * @param pageable standard paging parameters in the request
+     * @return ResponseEntity which contains an Pageable list of all free cars
+     * and Response Code 200 on success.
+     */
+    @GetMapping(path = "/free")
+    public @ResponseBody
+    ResponseEntity<Page<Car>> getAllFreeCars(@PageableDefault final Pageable pageable) {
+        return carService.getAllFreeCars(pageable);
+    }
+
+    /**
+     * Returns all cars from the database in a predefined ViewObject format.
+     * <pre>
+     *     Method GET
+     *     URL /api/v1/car/allCarVo
+     * </pre>
+     * Sample call /api/v1/car/allCarVo
+     *
+     * @param pageable standard paging parameters in the request
+     * @return ResponseEntity which contains an Pageable list of all cars in a viewObject format
+     * and Response Code 200 on success.
+     */
+    @GetMapping(path = "/car/allCarVo")
+    public @ResponseBody
+    ResponseEntity<Page<CarViewObject>> getAllCarViewObject(@PageableDefault final Pageable pageable) {
+        return carService.getAllCarViewObject(pageable);
+    }
+
+    /**
+     * The method is used for inserting a new Car object into the database, defined in the body of
+     * the request in a carViewObject format.
+     * <pre>
+     *     Method PUT
+     *     URL /api/v1/car/carVO
+     * </pre>
+     * Sample call /api/v1/car/carVO
+     *
+     * @param carViewObject insertable car in ViewObject format
+     * @return ResponseEntity with Response Code 200 on success.
+     */
+    @PutMapping(path = "/car/carVO", consumes = "application/json")
+    public @ResponseBody
+    ResponseEntity insertNewCarFromViewObject(@RequestBody final CarViewObject carViewObject) {
```

Comment:
What is the exact intention with this service? Also, Vo should not be exposed from the application. The car is just car for the user, internal representation details should not be leaked.